docs/writing-stories/mocking-data-and-modules/mocking-modules.mdx

```diff
@@ -5,17 +5,215 @@ sidebar:
+
+There are some key details to keep in mind when using the `sb.mock` utility:
+
+- You can register both local modules (e.g., `../lib/session.ts`) and packages in `node_modules` (e.g., `uuid`).
+- You can only register mocked modules in your project-level configuration: `.storybook/preview.js|ts`. This ensures consistent and performant mocking across all stories in your project. You can modify the behavior of these modules in your stories, but you cannot register them directly in the story files.
+- You must pass the full relative path, without aliasing and including the file extension, when registering a mock for a local module. For packages in `node_modules`, you can use the package name without the file extension.
+- If you are using Typescript, you can wrap the module path in `import()` to ensure the module is correctly resolved and typed. For example, `sb.mock(import('../lib/session.ts'))`.
+- If you are using the [Webpack builder](../../builders/webpack.mdx), you can only automatically mock `node_module` packages that have ESModules (ESM) entry points. If a module has both CommonJS (CJS) and ESM entry points, Webpack doesn't correctly resolve the ESM entry and it cannot be mocked. Webpack users can still mock CJS `node_module` packages by providing a [mock file](#mock-files).
+
+#### Spy-only
+
+For most cases, you should register a mocked module as spy-only, by setting the `spy` option to `true`. This leaves the original module's functionality intact, while still allowing you to modify the behavior if needed and make assertions in your tests.
+
+For example, if you want to spy on the `getUserFromSession` function and the `uuidv4` function from the `uuid` package, you can call the `sb.mock` utility function in your `.storybook/preview.js|ts` file:
+
+{/* TODO: Snippetize */}
+```ts title=".storybook/preview.ts"
+import { sb } from 'storybook/test';
+
+// 👇 Automatically spies on all exports from the `lib/session` local module
+sb.mock('../lib/session.ts', { spy: true });
+// 👇 Automatically spies on all exports from the `uuid` package in `node_modules`
+sb.mock('uuid', { spy: true });
+
+// ...rest of the file
+```
+
+You can then [control the behavior of these modules](#using-automocked-modules-in-stories) and make assertions about them in your stories, such as checking if a function was called or what arguments it was called with.
+
+#### Automatically mocked modules
+
+For cases where you need to prevent the original module's functionality from ever executing, set the `spy` option to `false` (or omit it, because that is the default value). This will automatically replace all exports from the module with [Vitest mock functions](https://vitest.dev/api/mock.html), allowing you to control their behavior and make assertions while being certain that the original functionality never runs.
```

Comment:
Even with `spy: false` the original module will still be evaluated, although all functions will return undefined. It's module and it's dependencies are still evaluated, which can be annoying if dependencies will crash in the browser for example.

The only way to make sure the original code never runs, is by writing a mock file in the __mocks__ directory.